src/zope/interface/adapter.py

```diff
@@ -119,6 +167,52 @@ def _createLookup(self):
         for name in self._delegated:
             self.__dict__[name] = getattr(self._v_lookup, name)
 
+    # Hooks for subclasses to define the types of objects used in
+    # our data structures.
+    # These have to be documented in the docstring, instead of local
+    # comments, because autodoc ignores the comment and just writes
+    # "alias of list"
+    _sequenceType = list
+    _leafSequenceType = tuple
+    _mappingType = dict
+    _providedType = dict
+
+    def _addValueToLeaf(self, existing_leaf_sequence, new_item):
+        """
+        Add the value *new_item* to the *existing_leaf_sequence*, which may
+        be ``None``.
+
+        If *existing_leaf_sequence* is not *None*, it will be an instance
+        of `_leafSequenceType`.
+
+        This method returns the new value to be stored. It may mutate the
+        sequence in place if it was not None and the type is mutable, but
+        it must also return it.
+
+        Subclasses that redefine `_leafSequenceType` should override this method.
+
+        .. versionadded:: 5.3.0
+        """
+        if existing_leaf_sequence is None:
+            return (new_item,)
+        return existing_leaf_sequence + (new_item,)
```

Comment:
I do not know if performance is an issue here, but concatenating tuples generates a new object which might have some impact on performance

Reply:
True! But: 

(a) that's what the existing code did, so there shouldn't be a regression (other than the slight cost of the added method call); and 

(b) even if registering new utilities/adapters is on a critical fast path of an application (seems doubtful to me, but assumptions like that can obviously be wrong!), this is probably the least part of that, when considering actually creating the component, testing whether it needs to be added to the subscribers sequence, emitting the event about the registration, etc.